caddyhttp/proxy/reverseproxy.go

```diff
@@ -254,6 +254,10 @@ func (rp *ReverseProxy) copyResponse(dst io.Writer, src io.Reader) {
 
 func copyHeader(dst, src http.Header) {
 	for k, vv := range src {
+		// some predefined headers (e.g. MIME plugin), skip.
+		if _, ok := dst[k]; ok {
+			continue
+		}
```

Comment:
I think we could observe these headers from proxy, but if we have predefined some of them, they will be overwritten, is it a problem (user may think its plugin not works, e.g. MIME)? Maybe we could clarify this in documents.

Also I think proxy directive is a proper place.